src/DynamoCore/Graph/Nodes/CodeBlockUtils.cs

```diff
@@ -71,21 +70,32 @@ internal static string NormalizeLineBreaks(string text)
         /// all variables defined in nested Statement objects.</param>
         /// <returns>Returns a list of lists of variables defined by the given 
         /// set of Statement objects.</returns>
-        /// 
         internal static IEnumerable<IEnumerable<string>> GetStatementVariables(
-            IEnumerable<Statement> statements, bool onlyTopLevel)
+            IEnumerable<Statement> statements)
         {
             if (statements == null)
-                throw new ArgumentNullException("statements");
+                throw new ArgumentNullException(nameof(statements));
 
-            var definedVariables = new List<List<string>>();
-            foreach (var statement in statements)
-            {
-                definedVariables.Add(Statement.GetDefinedVariableNames(
-                    statement, onlyTopLevel));
-            }
+            return statements.Select(Statement.GetDefinedVariableNames).ToList();
```

Comment:
If you're calling `ToList()` and the method is `internal`, why not simply make the return argument `List`? The `IEnumerable` abstraction actually loses valuable information for the caller.